x-pack/plugins/ml/server/lib/alerts/register_jobs_monitoring_rule_type.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { KibanaRequest } from 'kibana/server';
+import { ML_ALERT_TYPES } from '../../../common/constants/alerts';
+import { PLUGIN_ID } from '../../../common/constants/app';
+import { MINIMUM_FULL_LICENSE } from '../../../common/license';
+import {
+  anomalyDetectionJobsHealthRuleParams,
+  AnomalyDetectionJobsHealthRuleParams,
+} from '../../routes/schemas/alerting_schema';
+import { RegisterAlertParams } from './register_ml_alerts';
+import {
+  ActionGroup,
+  AlertInstanceContext,
+  AlertInstanceState,
+  AlertTypeState,
+} from '../../../../alerting/common';
+
+export type AnomalyDetectionJobsHealthAlertContext = {
+  jobIds: string[];
+  message: string;
+} & AlertInstanceContext;
+
+export const ANOMALY_DETECTION_JOB_REALTIME_ISSUE = 'anomaly_detection_realtime_issue';
+
+export type AnomalyDetectionJobRealtimeIssue = typeof ANOMALY_DETECTION_JOB_REALTIME_ISSUE;
+
+export const REALTIME_ISSUE_DETECTED: ActionGroup<AnomalyDetectionJobRealtimeIssue> = {
+  id: ANOMALY_DETECTION_JOB_REALTIME_ISSUE,
+  name: i18n.translate('xpack.ml.jobsHealthAlertingRule.actionGroupName', {
+    defaultMessage: 'Real-time issue detected',
+  }),
+};
+
+export function registerJobsMonitoringRuleType({
+  alerting,
+  mlServicesProviders,
+  logger,
+}: RegisterAlertParams) {
+  alerting.registerType<
+    AnomalyDetectionJobsHealthRuleParams,
```

Comment:
[This PR](https://github.com/elastic/kibana/pull/101896/files) just got merged so you will need to update the list of templated types. Suggest using `never` for now until you need to make use of saved object references. Example of the change to the other ML rule type [here](https://github.com/elastic/kibana/pull/101896/files#diff-71fccaf2dd3c66b6270dc3bf286761d17d83648677044bdbcda547f114293f96)

Reply:
thanks for the heads-up @ymao1, updated types in https://github.com/elastic/kibana/pull/106084/commits/23e74e650640e5a01a831675669db24eca226d9e